### PR TITLE
Add ability to change aes password

### LINKED ISSUE #2221: Allow the ability to update passwords
Currently in the code base we have two passwords 

1. the aes password used to encrypt the wallet
2. the bip39 mnemonic password 

We need to allow users to change these passwords

Related to #972

### PR DESCRIPTION
**Pull request by benthecarman**
Fixes #2221

Needed to fix `AesPassword`'s `toStringSensitive` as well as allow the console cli to output null json

### CODE REVIEW

File: app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala [hunk 1]
```diff
@@ -892,6 +896,44 @@ object ConsoleCli {
                 case other => other
               }))
         ),
+      cmd("walletpassphrasechange")
```
Comment (Christewart) [inline app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala]: This is really `keymanagerpassphrasechange` no?

File: key-manager-test/src/test/scala/org/bitcoins/keymanager/WalletStorageTest.scala [hunk 2]
```diff
@@ -89,6 +89,85 @@ class WalletStorageTest extends BitcoinSWalletTest with BeforeAndAfterEach {
...
+        WalletStorage.decryptMnemonicFromDisk(seedPath, None)
+      read match {
+        case Right(readMnemonic) =>
+          assert(writtenMnemonic.mnemonicCode == readMnemonic.mnemonicCode)
+          // Need to compare using getEpochSecond because when reading an epoch second
+          // it will not include the milliseconds that writtenMnemonic will have
+          assert(
+            writtenMnemonic.creationTime.getEpochSecond == readMnemonic.creationTime.getEpochSecond)
+        case Left(err) => fail(err.toString)
+      }
```
Comment (Christewart) [inline key-manager-test/src/test/scala/org/bitcoins/keymanager/WalletStorageTest.scala:168]: I think we need some "negative" test cases. The obvious one to me is

```scala

it must "failed to change the passphrase if the old passphrase is incorrect" in { ... }

```

File: key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala [hunk 3]
```diff
@@ -304,6 +304,33 @@ object WalletStorage {
...
 
+  def changeAesPassword(
+      seedPath: Path,
+      oldPasswordOpt: Option[AesPassword],
+      newPasswordOpt: Option[AesPassword]): MnemonicState = {
+    decryptMnemonicFromDisk(seedPath, oldPasswordOpt) match {
+      case Left(err) => sys.error(err.toString)
+      case Right(decrypted) =>
+        val fileName = seedPath.getFileName.toString
+        val backup = seedPath.getParent.resolve(fileName + ".backup")
```
Comment (Christewart) [inline key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala:317]: Is there any way we can test this backup logic?

File: key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala [hunk 4]
```diff
@@ -304,6 +304,33 @@ object WalletStorage {
...
+
+        val toWrite = newPasswordOpt match {
+          case Some(pass) =>
+            decrypted.encrypt(pass)
+          case None =>
+            decrypted
+        }
+
+        Try(writeMnemonicToDisk(seedPath, toWrite)) match {
+          case Failure(exception) => throw exception
```
Comment (Christewart) [inline key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala]: We should probably add a `log.warn()` or `log.error()` saying that the backup seed can be found at the backup file path.

File: key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala [hunk 5]
```diff
@@ -304,6 +304,33 @@ object WalletStorage {
     }
   }
 
+  def changeAesPassword(
+      seedPath: Path,
+      oldPasswordOpt: Option[AesPassword],
+      newPasswordOpt: Option[AesPassword]): MnemonicState = {
+    decryptMnemonicFromDisk(seedPath, oldPasswordOpt) match {
```
Comment (Christewart) [inline key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala:312]: We should probably have a `log.info()` indicating we are attempting to change the password of the seed.

File: key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala (hunk 4 above)
Comment (Christewart) [reply key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala]: Do we want to move the backup file back to the original seed in this case? I'm 50/50 on that. It seems dangerous in the sense that it makes it easier to "silently" fail changing the password. 

However you may want to write application logic to handle this case and revert to using the `oldPasswordOpt` without downtime. Thoughts?

File: app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala (hunk 1 above)
Comment (Christewart) [reply app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala]: Same thing applies to all commands I think

**Review by Christewart — CHANGES_REQUESTED** [review]

File: key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala (hunk 3 above)
Comment (benthecarman) [reply key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala:317]: No :/

File: key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala (hunk 4 above)
Comment (benthecarman) [reply key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala]: I think it's fine as is. Likely if we need the backup file something went wrong outside our control, we should probably just leave the file and not touch it

**Review by Christewart — APPROVED** [review]

**Pull request merged**